### ISSUE TITLE
[Bug] Two bugs related to the event system

### ISSUE THREAD

**Issue by CatJet** [Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker#596]
(Note: This initial comment describes the first bug, the comment I made below the second one)
### Describe the bug

Triggering an event that's linked to a function that calls `Explosion()` while the listener script has `handleCommand()` crashes the game.

### Steps to reproduce the behavior

1. Make a new scene and add a script node with the following script attached to it:
``` lua
function init()
	RegisterListenerTo("Definitely Harmless Event", "Crash")
end

function tick()
	if InputPressed("h") then
		TriggerEvent("Definitely Harmless Event")
	end
end

function Crash()
	Explosion(Vec(), 1)
end

function handleCommand(cmd)
	DebugPrint(cmd)
end

-- handleCommand could have literally nothing in it and it would still cause a crash, like so:
-- function handleCommand() end
```
2. Add a simple voxbox to stand on and `main.lua` and `sandbox.lua` to the scene so you have tools to test with.
3. Play the scene. Shoot with the gun or rocket launcher and notice the debug text indicating that `handleCommand` is working normally
4. Press h to trigger the event, which causes the crash

### Expected behavior

The event would have called `Explosion` normally, the explosion would've been detected by `handleCommand` and the game would continue running normally.

### Environment

- Teardown Version: 1.6.0
- OS: Linux
- GPU: Intel UHD 630
- Graphics driver version: `glxinfo -B` says `Version: 24.0.9` under `Extended renderer info` ¯\\\_(ツ)\_/¯
- CPU: Intel Core i5-10400

### Additional context

Further testing I did:
- If the `handleCommand` function isn't defined like in the script below, the event works normally and doesn't crash the game:
``` lua
-- Doesn't crash
function init()
	RegisterListenerTo("Definitely Harmless Event", "Crash")
end

function tick()
	if InputPressed("h") then
		TriggerEvent("Definitely Harmless Event")
	end
end

function Crash()
	Explosion(Vec(), 1)
end
```

- Calling `Explosion` directly in the `InputPressed("h")` clause like in the script below won't crash the game and causes handleCommand to print out the right string (`"explosion 1.0 0.0 0.0 0.0"`):
``` lua
-- Doesn't crash
function tick()
	if InputPressed("h") then
		Explosion(Vec(), 1)
	end
end

function handleCommand(cmd)
	DebugPrint(cmd)
end
```

More interesting stuff about this bug when multiple scripts are used:
- This crashes:
``` lua
-- file 1
function init()
 	RegisterListenerTo("Event 1", "Function1")
end

function tick()
	if InputPressed("h") then
		TriggerEvent("Event 1")
	end
end

function Function1()
 	TriggerEvent("Event 2")
end

function handleCommand() end
```
``` lua
-- file 2
function init()
	RegisterListenerTo("Event 2", "Function2")
end

function Function2()
	Explosion(Vec(), 1)
end
```
- This doesn't crash:
``` lua
-- file 1
function tick()
	if InputPressed("h") then
		TriggerEvent("Event 2")
	end
end

function handleCommand() end
```
``` lua
-- file 2
function init()
	RegisterListenerTo("Event 2", "Function2")
end

function Function2()
	Explosion(Vec(), 1)
end
```
- While this crashes again:
``` lua
-- file 1
function tick()
	if InputPressed("h") then
		TriggerEvent("Event 2")
	end
end
```
``` lua
-- file 2
function init()
	RegisterListenerTo("Event 2", "Function2")
end

function Function2()
	Explosion(Vec(), 1)
end

function handleCommand() end
```

**Comment by CatJet** [comment]
I also found an adjacent bug with the event system that causes a crash if a script listened to 2 different events being triggered in the same tick.

1. Make a scene with the following two script nodes inside:
``` lua
-- file 1
function init()
	RegisterListenerTo("Event 2", "Function2")
end

function tick()
	if InputPressed('h') then
		TriggerEvent("Event 1")
	end
end

function Function2()
	TriggerEvent("Event 3")
end
```
``` lua
-- file 2
function init()
	RegisterListenerTo("Event 1", "Function1")
	RegisterListenerTo("Event 3", "Function3")
end

function Function1()
	TriggerEvent("Event 2")
end

function Function3()
	DebugPrint("At this point, the game will crash")
end
```
2. Press `h` to trigger the events, which cause the game to crash

**Comment by CatJet** [comment]
Just noticed that the game generates crash logs, so I snagged two of them (one for each bug) and attached 'em here
[2 Event-related Crashes.zip](https://github.com/user-attachments/files/16006084/2.Event-related.Crashes.zip)

**Comment by Gregory-Gregory** [comment]
@CatJet hi!

Thank you for reporting these issues and attached the crash logs. I have forwarded both issues to the QA team for further investigation.

**Comment by YuLun-bili** [comment]
resolved in 1.6.0 experimental
